pontoon/translate/static/js/translate.es6

```diff
@@ -0,0 +1,499 @@
+/**
+ * Model for Entities, which represent a single string to be translated.
+ */
+class Entity {
+  constructor(data) {
+    $.extend(this, data);
+  }
+
+  /**
+   * Return the first approved translation in this.translations, or null
+   * if there are none.
+   */
+  get approvedTranslation() {
+    for (let translation of this.translations) {
+      if (translation.approved) {
+        return translation;
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Return the current translation status.
+   */
+  get status() {
+    if (this.approvedTranslation) {
+      return 'approved';
+    } else if (this.translations.some((t) => t.fuzzy)) {
+      return 'fuzzy'
+    } else if (this.translations.length > 0) {
+      return 'translated';
+    } else {
+      return '';
+    }
+  }
+
+  updateTranslation(string, localeCode) {
+    let url = `/entity/${this.pk}/translations/${localeCode}/`;
+    return $.post(url, {string: string}).then((translations) => {
+      return $.extend(this, {translations: translations})
+    });
+  }
+
+  /**
+   * Fetch all the entities for the given project and locale.
+   */
+  static fetchAll(projectSlug, localeCode) {
+    let url = `/project/${projectSlug}/locale/${localeCode}/entities`;
+    return $.get(url).then((entities) => {
+      return entities.map((data) => new Entity(data));
+    });
+  }
+}
+
+
+/**
+ * Base class for React Components.
+ */
+class PontoonComponent extends React.Component {
+  /** Shorthand for getting a jQuery object for a ref. */
+  $ref(refName) {
+    return $(React.findDOMNode(this.refs[refName]));
```

Comment:
I used it at one point but then got rid of that use, I could've sworn the commit where I just removed this base class completely was in this branch. Whoops!